Build fails in MSYS2 CLANG64 environment

Fixing this is a gateway for the [clangarm64](https://repo.msys2.org/mingw/clangarm64/) environement (under construction) that'll facilitate Windows on Arm (WoA) deployment.

`cmake -G"MSYS Makefiles" -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF ../`

fails w/

```
-- Check for working C compiler: C:/msys64/clang64/bin/gcc.exe - broken
CMake Error at C:/msys64/clang64/share/cmake-3.21/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "C:/msys64/clang64/bin/gcc.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/msys64/home/milos/rawspeed/build/CMakeFiles/CMakeTmp

    Run Build Command(s):C:/msys64/usr/bin/make.exe -f Makefile cmTC_8cb79/fast && /usr/bin/make  -f CMakeFiles/cmTC_8cb79.dir/build.make CMakeFiles/cmTC_8cb79.dir/build
    make[1]: Entering directory '/home/milos/rawspeed/build/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_8cb79.dir/testCCompiler.c.obj
    /C/msys64/clang64/bin/gcc.exe    -MD -MT CMakeFiles/cmTC_8cb79.dir/testCCompiler.c.obj -MF CMakeFiles/cmTC_8cb79.dir/testCCompiler.c.obj.d -o CMakeFiles/cmTC_8cb79.dir/testCCompiler.c.obj -c /C/msys64/home/milos/rawspeed/build/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_8cb79.exe
    /C/msys64/clang64/bin/cmake.exe -E rm -f CMakeFiles/cmTC_8cb79.dir/objects.a
    /C/msys64/clang64/bin/llvm-ar.exe qc CMakeFiles/cmTC_8cb79.dir/objects.a "CMakeFiles/cmTC_8cb79.dir/testCCompiler.c.obj"
    /C/msys64/clang64/bin/gcc.exe  -Wl,--as-needed  -Wl,--whole-archive CMakeFiles/cmTC_8cb79.dir/objects.a -Wl,--no-whole-archive -o cmTC_8cb79.exe -Wl,--out-implib,libcmTC_8cb79.dll.a -Wl,--major-image-version,0,--minor-image-version,0  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
    lld: error: unknown argument: --as-needed
    gcc: error: linker command failed with exit code 1 (use -v to see invocation)
```

Maybe add the MINGW + Clang check in addition to APPLE [here](https://github.com/darktable-org/rawspeed/blob/95d1171398897af1518023c4fa44d6d581c370ca/cmake/compiler-flags.cmake#L45)?

With that patched it goes a bit further but still fails:

```
-- Trying to query CPU L1d cache line size
CMake Error at cmake/Modules/cpu-cache-line-size.cmake:15 (message):
  Failed to query CPU L1d cache line size:

  Change Dir: C:/msys64/home/milos/rawspeed/build/CMakeFiles/CMakeTmp



  Run Build Command(s):C:/msys64/usr/bin/make.exe -f Makefile cmTC_606a0/fast
  && /usr/bin/make -f CMakeFiles/cmTC_606a0.dir/build.make
  CMakeFiles/cmTC_606a0.dir/build

  make[1]: Entering directory
  '/home/milos/rawspeed/build/CMakeFiles/CMakeTmp'

  Building CXX object CMakeFiles/cmTC_606a0.dir/cpu-cache-line-size.cpp.obj

  /C/msys64/clang64/bin/g++.exe -std=c++17 -Wall -Wextra -Weverything
  -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion
  -Wno-covered-switch-default -Wno-deprecated -Wno-double-promotion
  -Wno-exit-time-destructors -Wno-global-constructors
  -Wno-gnu-zero-variadic-macro-arguments -Wno-old-style-cast -Wno-padded
  -Wno-switch-enum -Wno-unused-macros -Wno-unused-parameter -Wno-weak-vtables
  -Wno-used-but-marked-unused -Wno-nonportable-system-include-path
  -Wextra-semi -Wno-format -Wframe-larger-than=4096 -Wlarger-than=32768
  -Werror -std=c++17 -MD -MT
  CMakeFiles/cmTC_606a0.dir/cpu-cache-line-size.cpp.obj -MF
  CMakeFiles/cmTC_606a0.dir/cpu-cache-line-size.cpp.obj.d -o
  CMakeFiles/cmTC_606a0.dir/cpu-cache-line-size.cpp.obj -c
  /C/msys64/home/milos/rawspeed/cmake/Modules/cpu-cache-line-size.cpp

  C:/msys64/home/milos/rawspeed/cmake/Modules/cpu-cache-line-size.cpp:39:34:
  error: zero as null pointer constant
  [-Werror,-Wzero-as-null-pointer-constant]

    GetLogicalProcessorInformation(0, &buffer_size);
                                   ^
                                   nullptr

  1 error generated.
```

Note 1: gcc.exe and g++.exe are just copies of clang.exe (and clang++.exe) in CLANG64

Note 2: [this CLANG64 environment's toolchain is significantly different to the MINGW64 one](https://www.msys2.org/docs/environments/), where just switching compiler (CC=clang CXX=clang++) also fails on the cash line query. The default GCC toolchain in MINGW64 works obviously.

Clang version in both CLANG64 and MINGW64 is currently 13.

See also this [example for including more MSYS2 environments in the CI matrix](https://github.com/msys2/setup-msys2/blob/master/examples/cmake.yml).

--- a/cmake/compiler-flags.cmake
+++ b/cmake/compiler-flags.cmake
@@ -42,7 +42,7 @@ set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
-IF(NOT APPLE)
+IF(NOT APPLE AND NOT WIN32)
   set(linkerflags "-Wl,--as-needed")
 ELSE()
   set(linkerflags "")

